test/shape_test.cpp

```diff
@@ -160,6 +160,34 @@ TEST_CASE(test_shape_dynamic_compares)
     EXPECT(ss0.str() != ss3.str());
 }
 
+TEST_CASE(dynamic_shape_element_space)
+{
+    migraphx::shape s{migraphx::shape::float_type, {{1, 10}, {3, 20, {3}}}};
+    EXPECT(s.element_space() == 200);
+}
+
+TEST_CASE(dynamic_shape_element_space_overflow0)
+{
+    std::size_t max_val = std::numeric_limits<std::size_t>::max();
+    migraphx::shape s{migraphx::shape::float_type, {{0, max_val}, {0, max_val}}};
+    EXPECT(s.element_space() == max_val);
+}
+
+TEST_CASE(dynamic_shape_element_space_overflow1)
+{
+    std::size_t max_val   = std::numeric_limits<std::size_t>::max();
+    std::size_t large_val = max_val / 10;
+    migraphx::shape s{migraphx::shape::float_type, {{0, large_val}, {0, large_val}}};
+    EXPECT(s.element_space() == max_val);
```

Comment:
If we overflow do we not expect to error out/ signal an overflow as a warning/error?